### DISCUSSION TITLE
How can capture the close event of finplot?

### DISCUSSION THREAD

**Discussion by hwjung92** [highfestiva/finplot#163] [General]
The below function cannot capture the event.
How can capture the close event of finplot?

def closeEvent(self, event):
        print('Window closed')

**Comment by highfestiva** [comment]
Possibly `finplot.FinWindow.close`?

**Reply by hwjung92** [reply]
That is not work.
I mean when I click the window close (X) at the right-top, how can I receive the event in the fplt?
I want to clear all fplt data when closing a fin-plot window.

**Reply by highfestiva** [reply]
If you use `finplot.show()` some clearing is done for you when the window is closed. Otherwise replace `finplot.FinWindow.close` with a method of your choosing.

**Reply by hwjung92** [reply]
In my application, if finplot is not cleared well, the main program has a conflict for using Timer.
matplotlib provide the below function to receive the close event.
Do you provide it like a below code?


=======================================================
class Plotter:
    def __init__(self):
        self.fig = plt.figure()
        self.fig.canvas.mpl_connect('close_event', self.dispose)
        plt.plot(1, 2, 'r*')
        plt.show()
        print "done with plotter"

    def dispose(self, event):
        plt.close('all')
        print "disposed"

**Reply by highfestiva** [reply]
This seems to work.

```python
#!/usr/bin/env python3

import finplot as fplt
import numpy as np
import pandas as pd

def update():
    dates = pd.date_range('01:00', '01:00:01.200', freq='1ms')
    prices = pd.Series(np.random.random(len(dates))).rolling(30).mean() + 4
    ax.reset()
    fplt.plot(dates, prices, ax=ax)
    fplt.refresh()

ax = fplt.create_plot(maximize=False)
fplt.timer_callback(update, 0.02)
fplt.show()
```

Not sure what you're problem is, since you're not supplying.
